### ISSUE TITLE
TypeError: Couldn't build proto file into descriptor pool!
Invalid proto descriptor for file "plugins/shuffle/protobuf/message.proto":
  Address.address: "Address.address" is already defined in file "messages.proto".
  Address: "Address" is already defined in file "messages.proto".
  Message.address: "Address" seems to be defined in "messages.proto", which is not imported by "plugins/shuffle/proto...

### ISSUE THREAD

**Issue by ProfessorVenkman** [Electron-Cash/Electron-Cash#1227]
Crash Report
============

This crash report was reported through the automatic crash reporting system 🤖

Traceback
--------------

```Python traceback
  File "site-packages\electron_cash-4.0.0-py3.5.egg\electroncash_gui\qt\main_window.py", line 82, in onPress
  File "site-packages\electron_cash-4.0.0-py3.5.egg\electroncash_gui\qt\main_window.py", line 3115, in cashshuffle_icon_leftclick
  File "site-packages\electron_cash-4.0.0-py3.5.egg\electroncash_gui\qt\main_window.py", line 3141, in toggle_cashshuffle
  File "site-packages\electron_cash-4.0.0-py3.5.egg\electroncash\plugins.py", line 228, in enable_internal_plugin
  File "site-packages\electron_cash-4.0.0-py3.5.egg\electroncash\plugins.py", line 154, in get_internal_plugin
  File "site-packages\electron_cash-4.0.0-py3.5.egg\electroncash\plugins.py", line 177, in load_internal_plugin
  File "<frozen importlib._bootstrap_external>", line 418, in _check_name_wrapper
  File "<frozen importlib._bootstrap_external>", line 841, in load_module
  File "<frozen importlib._bootstrap_external>", line 700, in load_module
  File "<frozen importlib._bootstrap>", line 268, in _load_module_shim
  File "<frozen importlib._bootstrap>", line 693, in _load
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 697, in exec_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
  File "C:\Users\jbrac\AppData\Local\Temp\_MEI102922\electroncash_plugins\shuffle\qt.py", line 47, in <module>
    from electroncash_plugins.shuffle.client import BackgroundShufflingThread, ERR_SERVER_CONNECT, ERR_BAD_SERVER_PREFIX, MSG_SERVER_OK
  File "C:\Users\jbrac\AppData\Local\Temp\_MEI102922\electroncash_plugins\shuffle\client.py", line 17, in <module>
    from .messages import Messages, AbortProtocol
  File "C:\Users\jbrac\AppData\Local\Temp\_MEI102922\electroncash_plugins\shuffle\messages.py", line 1, in <module>
    from . import message_pb2 as message_factory
  File "C:\Users\jbrac\AppData\Local\Temp\_MEI102922\electroncash_plugins\shuffle\message_pb2.py", line 23, in <module>
    serialized_pb=_b('\n&plugins/shuffle/protobuf/message.proto\"@\n\x06Signed\x12\x17\n\x06packet\x18\x01 \x01(\x0b\x32\x07.Packet\x12\x1d\n\tsignature\x18\x02 \x01(\x0b\x32\n.Signature\"\xc6\x01\n\x06Packet\x12\x0f\n\x07session\x18\x01 \x01(\x0c\x12\x0e\n\x06number\x18\x02 \x01(\r\x12\"\n\x08\x66rom_key\x18\x03 \x01(\x0b\x32\x10.VerificationKey\x12 \n\x06to_key\x18\x04 \x01(\x0b\x32\x10.VerificationKey\x12\x15\n\x05phase\x18\x05 \x01(\x0e\x32\x06.Phase\x12\x19\n\x07message\x18\x06 \x01(\x0b\x32\x08.Message\x12#\n\x0cregistration\x18\x07 \x01(\x0b\x32\r.Registration\"\x16\n\x05\x43oins\x12\r\n\x05\x63oins\x18\x01 \x03(\t\"9\n\nSignatures\x12\x0c\n\x04utxo\x18\x01 \x01(\t\x12\x1d\n\tsignature\x18\x02 \x01(\x0b\x32\n.Signature\"\xf8\x01\n\x07Message\x12\x19\n\x07\x61\x64\x64ress\x18\x01 \x01(\x0b\x32\x08.Address\x12\x1b\n\x03key\x18\x02 \x01(\x0b\x32\x0e.EncryptionKey\x12\x13\n\x04hash\x18\x03 \x01(\x0b\x32\x05.Hash\x12\x1f\n\nsignatures\x18\x04 \x03(\x0b\x32\x0b.Signatures\x12\x0b\n\x03str\x18\x05 \x01(\t\x12\x15\n\x05\x62lame\x18\x06 \x01(\x0b\x32\x06.Blame\x12$\n\x06inputs\x18\x07 \x03(\x0b\x32\x14.Message.InputsEntry\x1a\x35\n\x0bInputsEntry\x12\x0b\n\x03key\x18\x01 \x01(\t\x12\x15\n\x05value\x18\x02 \x01(\x0b\x32\x06.Coins:\x02\x38\x01\"\x1a\n\x07\x41\x64\x64ress\x12\x0f\n\x07\x61\x64\x64ress\x18\x01 \x01(\t\"K\n\x0cRegistration\x12\x0e\n\x06\x61mount\x18\x01 \x01(\x04\x12\x1a\n\x04type\x18\x02 \x01(\x0e\x32\x0c.ShuffleType\x12\x0f\n\x07version\x18\x03 \x01(\x04\"\x1e\n\x0fVerificationKey\x12\x0b\n\x03key\x18\x01 \x01(\t\"\x1c\n\rEncryptionKey\x12\x0b\n\x03key\x18\x01 \x01(\t\",\n\rDecryptionKey\x12\x0b\n\x03key\x18\x01 \x01(\t\x12\x0e\n\x06public\x18\x02 \x01(\t\"\x14\n\x04Hash\x12\x0c\n\x04hash\x18\x01 \x01(\x0c\"\x1e\n\tSignature\x12\x11\n\tsignature\x18\x01 \x01(\x0c\"\"\n\x0bTransaction\x12\x13\n\x0btransaction\x18\x01 \x01(\x0c\"\xb9\x01\n\x05\x42lame\x12\x17\n\x06reason\x18\x01 \x01(\x0e\x32\x07.Reason\x12!\n\x07\x61\x63\x63used\x18\x02 \x01(\x0b\x32\x10.VerificationKey\x12\x1b\n\x03key\x18\x03 \x01(\x0b\x32\x0e.DecryptionKey\x12!\n\x0btransaction\x18\x04 \x01(\x0b\x32\x0c.Transaction\x12\x19\n\x07invalid\x18\x05 \x01(\x0b\x32\x08.Invalid\x12\x19\n\x07packets\x18\x06 \x01(\x0b\x32\x08.Packets\"\x1a\n\x07Invalid\x12\x0f\n\x07invalid\x18\x01 \x01(\x0c\"(\n\x06Inputs\x12\x0f\n\x07\x61\x64\x64ress\x18\x01 \x01(\t\x12\r\n\x05\x63oins\x18\x02 \x03(\t\"\"\n\x07Packets\x12\x17\n\x06packet\x18\x01 \x03(\x0b\x32\x07.Signed*\x90\x01\n\x05Phase\x12\x08\n\x04NONE\x10\x00\x12\x10\n\x0c\x41NNOUNCEMENT\x10\x01\x12\x0b\n\x07SHUFFLE\x10\x02\x12\r\n\tBROADCAST\x10\x03\x12\x16\n\x12\x45QUIVOCATION_CHECK\x10\x04\x12\x0b\n\x07SIGNING\x10\x05\x12\x1f\n\x1bVERIFICATION_AND_SUBMISSION\x10\x06\x12\t\n\x05\x42LAME\x10\x07*$\n\x0bShuffleType\x12\x0b\n\x07\x44\x45\x46\x41ULT\x10\x00\x12\x08\n\x04\x44UST\x10\x01*\xc6\x01\n\x06Reason\x12\x15\n\x11INSUFFICIENTFUNDS\x10\x00\x12\x0f\n\x0b\x44OUBLESPEND\x10\x01\x12\x17\n\x13\x45QUIVOCATIONFAILURE\x10\x02\x12\x12\n\x0eSHUFFLEFAILURE\x10\x03\x12!\n\x1dSHUFFLEANDEQUIVOCATIONFAILURE\x10\x04\x12\x14\n\x10INVALIDSIGNATURE\x10\x05\x12\x11\n\rMISSINGOUTPUT\x10\x06\x12\x08\n\x04LIAR\x10\x07\x12\x11\n\rINVALIDFORMAT\x10\x08\x62\x06proto3')
  File "site-packages\google\protobuf\descriptor.py", line 829, in __new__

TypeError: Couldn't build proto file into descriptor pool!
Invalid proto descriptor for file "plugins/shuffle/protobuf/message.proto":
  Address.address: "Address.address" is already defined in file "messages.proto".
  Address: "Address" is already defined in file "messages.proto".
  Message.address: "Address" seems to be defined in "messages.proto", which is not imported by "plugins/shuffle/protobuf/message.proto".  To use it here, please add the necessary import.

```

Reporter
------------

This issue was reported by 2 user(s):

| Electron Cash Version  | Python Version | Operating System  | Wallet Type  | Locale |
|---|---|---|---|---|
| 4.0.0  | 3.5.4 (v3.5.4:3f56838, Aug  8 2017, 02:07:06) [MSC v.1900 32 bit (Intel)] | Windows-10-10.0.17134 | standard | en_US | 
| 4.0.1  | 3.5.6 (default, Sep 20 2018, 11:49:45)  [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)] | Darwin-17.7.0-x86_64-i386-64bit | imported_privkey | en_US | 

Additional Information
------------------------

> Tried to enable cash shuffle.
> 
> New standard wallet. 
> 
> No balance.

---

> Uh oh. Calin here. There is a conflict between CashShuffle's messages.proto and keepkey's. WTF

---

**Comment by cculianu** [comment]
There's something funny going on here. I can't reproduce this on Windows 10 at all. Not sure what's going on in this guy's setup .. the error messages don't make any sense at all.

**Comment by cculianu** [comment]
I wonder if this is a conflict with the old cashshuffle...? Weird.

**Comment by cculianu** [comment]
This was a  bug in protobuf, basically.  Fixed in this commit: 585675e023c9f0aa64ce7d2fad0c87c6b50ebddb
